src/events/slices/events.ts

```diff
@@ -0,0 +1,116 @@
+import { createAsyncThunk, createEntityAdapter, createSlice, SerializedError, unwrapResult } from '@reduxjs/toolkit';
+import { State } from 'core/redux/Store';
+import { getEvent, getEvents, IEventAPIParameters } from 'events/api/events';
+import { EventTypeEnum, IEvent } from 'events/models/Event';
+import { DateTime } from 'luxon';
+
+const eventsAdapter = createEntityAdapter<IEvent>({
+  sortComparer: (eventA, eventB) => {
+    return Number(DateTime.fromISO(eventA.event_start) > DateTime.fromISO(eventB.event_start));
+  },
+});
+
+export const eventSelectors = eventsAdapter.getSelectors<State>((state) => state.events);
+
+export const fetchEvents = createAsyncThunk('events/fetchMultiple', async (options?: IEventAPIParameters) => {
+  const events = await getEvents(options);
+  return events;
+});
+
+export const fetchEventById = createAsyncThunk('events/fetchById', async (eventId: number) => {
+  const events = await getEvent(eventId);
+  return events;
+});
+
+export const fetchEventList = createAsyncThunk('events/fetchList', async (_, { dispatch }) => {
+  const response = await dispatch(
+    fetchEvents({
+      event_end__gte: DateTime.local().toISODate(),
+      page_size: 10,
+    })
+  );
+  return unwrapResult(response);
+});
+
+export const fetchImageEvents = createAsyncThunk('events/fetchImageEvents', async (_, { dispatch }) => {
+  const getTypeEvents = async (types: EventTypeEnum[]) => {
+    return await dispatch(
+      fetchEvents({
+        event_end__gte: DateTime.local().toISODate(),
+        event_type: types,
+        page_size: 4,
+      })
+    );
+  };
+  // Separate the three column fetches to be able to present some columns even if 1 type of events fails.
+  const left = getTypeEvents([EventTypeEnum.BEDPRES]);
+  const middle = getTypeEvents([EventTypeEnum.KURS]);
+  const right = getTypeEvents([
+    EventTypeEnum.SOSIALT,
+    EventTypeEnum.UTFLUKT,
+    EventTypeEnum.EKSKURSJON,
+    EventTypeEnum.ANNET,
+  ]);
+  const responses = await Promise.all([left, right, middle]);
+  responses.forEach((payloadCallback) => unwrapResult(payloadCallback));
```

Comment:
Will this actually return the responses?
As I understand `forEach` it doesn't actually change the array?